core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -1581,21 +1581,15 @@ struct TestIsNaN {
     }
 #if !defined(KOKKOS_ENABLE_SYCL) && \
     !defined(KOKKOS_ENABLE_HIP)  // FIXME_SYCL, FIXME_HIP
-    if (isnan(static_cast<KE::half_t>(2.f))
-#if !defined(KOKKOS_ENABLE_CUDA)
-        || !isnan(quiet_NaN<KE::half_t>::value) ||
-        !isnan(signaling_NaN<KE::half_t>::value)
-#endif
-    ) {
+    if (isnan(static_cast<KE::half_t>(2.f)) ||
+        !isnan(static_cast<KE::half_t>(quiet_NaN<KE::half_t>::value)) ||
+        !isnan(static_cast<KE::half_t>(signaling_NaN<KE::half_t>::value))) {
```

Comment:
Does this work: 

```c++
KE::half_t val = quiet_NaN<KE::half_t>::value;
assert(val == quiet_NaN<KE::half_t>::value);
```

I would assume it does right?

Reply:
NaN does not compare equal to itself